shell/src/main/java/tachyon/shell/command/ChgrpCommand.java

```diff
@@ -51,18 +57,41 @@ protected int getNumOfArgs() {
 
   @Override
   public void run(String... args) throws IOException {
-    String group = args[0];
-    TachyonURI path = new TachyonURI(args[1]);
-    chgrp(path, group, false);
+    CommandLineParser parser = new BasicParser();
+    Options options = new Options();
+    // Add R option for chgrp recursively.
+    options.addOption("R", false, "recursively");
+    CommandLine cmd;
+    try {
+      cmd = parser.parse(options, args);
+    } catch (ParseException e) {
+      System.err.println("Unable to parse input args: " + e.getMessage());
+      return;
+    }
+
+    String group = cmd.getArgs()[0];
+    TachyonURI path = new TachyonURI(cmd.getArgs()[1]);
+
+    chgrp(path, group, cmd.hasOption("R"));
   }
 
   @Override
   public String getUsage() {
-    return "chgrp <group> <path>";
+    return "chgrp [-R] <group> <path>";
   }
 
   @Override
   public String getDescription() {
-    return "Changes the group of a file or directory specified by args.";
+    return "Changes the group of a file or directory specified by args.\n"
+        + "Specify -R to change the group recursively.";
+  }
+
+  @Override
+  public boolean validateArgs(String... args) {
```

Comment:
Instead of overriding `validateArgs` could this class communicate with the superclass via a `getOptions` method so that the superclass can do all parsing/validation? The `CommandLine` could be passed into the `run` method.